src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -250,6 +250,37 @@ public void NullBaseDirectoryNotAddedToKeyFileSearchPaths()
             AssertEx.Equal(ImmutableArray.Create<string>(), parser.KeyFileSearchPaths);
         }
 
+        [Fact]
+        [WorkItem(29252, "https://github.com/dotnet/roslyn/issues/29252")]
+        public void SdkPathArg()
+        {
+            var parentDir = Temp.CreateDirectory();
+
+            var sdkDir = parentDir.CreateDirectory("sdk");
+            var sdkPath = sdkDir.Path;
+
+            var parser = CSharpCommandLineParser.Default.Parse(new[] { $"-sdkPath:{sdkPath}" }, null, null);
+            AssertEx.Equal(ImmutableArray.Create(sdkPath), parser.ReferencePaths);
+        }
+
+        [Fact]
+        [WorkItem(29252, "https://github.com/dotnet/roslyn/issues/29252")]
+        public void SdkPathNoArg()
+        {
+            var parentDir = Temp.CreateDirectory();
+            var parser = CSharpCommandLineParser.Default.Parse(new[] { "file.cs", $"-out:{parentDir.Path}", "-sdkPath" }, parentDir.Path, null);
+            parser.Errors.Verify(Diagnostic(ErrorCode.ERR_SwitchNeedsString, arguments: new[] { "<text>", "-sdkPath" }).WithLocation(1, 1));
+        }
+
+        [Fact]
+        [WorkItem(29252, "https://github.com/dotnet/roslyn/issues/29252")]
+        public void SdkPathNull()
+        {
+            var parentDir = Temp.CreateDirectory();
+            var parser = CSharpCommandLineParser.Default.Parse(new[] { "file.cs", $"-out:{parentDir.Path}", "/sdkPath-" }, parentDir.Path, null);
+            AssertEx.Equal(ImmutableArray<string>.Empty, parser.ReferencePaths);
+        }
```

Comment:
Please add a test where there is both `/sdkPath <value>` and `/sdkPath-` to make sure the last one wins.